homeassistant/components/climate_scheduler/__init__.py

```diff
@@ -0,0 +1,142 @@
+"""
+Schedules temperature updates to the climate component.
+
+Schedule is read from 'schedule_persistance.json' in the
+configuration directory.
+
+Lovelace card to control it (WIP, in spanish):
+https://gist.github.com/alex-torregrosa/482f62fec60d892b60b9dfe73289cd6f
+
+Example component configuration:
+
+climate_scheduler:
+  target_climate: climate.home
+"""
+import logging
+import voluptuous as vol
+import json
+import os
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import dt
+from homeassistant.helpers.event import (
+    async_track_time_change)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import STATE_ON
+from homeassistant.core import callback
+
+
+_LOGGER = logging.getLogger(__name__)
+
+# The domain of your component. Equal to the filename of your component.
+DOMAIN = "climate_scheduler"
+ENTITY_ID = "climate_scheduler.main"
+CONF_CLIMATE = 'target_climate'
+ATTR_TIMES = "timetable"
+DEFAULT_TIMES = "[]"
+PERISSTENCE_FILE = 'schedule_persistance.json'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CLIMATE): cv.entity_id,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Setups the climate scheduler component."""
+    climate = config[DOMAIN].get(CONF_CLIMATE, None)
+    if climate is None:
+        _LOGGER.error("No climate provided.")
+    ClimateScheduler(hass, climate)
+    # Return boolean to indicate that initialization was successfully.
+    return True
+
+
+class ClimateScheduler(Entity):
+    """Representation of a climate scheduler entity."""
+
+    entity_id = ENTITY_ID
+
+    def __init__(self, hass, climate):
+        """Initialize the scheduler."""
+        self.hass = hass
+        self._state = STATE_ON
+        self.climate = climate
+        self.rules = []
+        self.unsubs = []
+        self.persistence_file = hass.config.path(PERISSTENCE_FILE)
```

Comment:
You can probably use https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/storage.py

Reply:
Ok, thanks! I didn't know it existed.